src/AIAPI/MainGame/GameApi.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+// using ActionGame;
+using BepInEx.Bootstrap;
+using KKAPI.Studio;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using Manager;
+using AIProject;
+
+namespace KKAPI.MainGame
+{
+    /// <summary>
+    /// Provides API for interfacing with the main game. It is useful mostly in the actual game, but some
+    /// functions will work outside of it (for example in FreeH).
+    /// </summary>
+    public static partial class GameAPI
+    {
+        private static readonly Dictionary<GameCustomFunctionController, string> _registeredHandlers = new Dictionary<GameCustomFunctionController, string>();
+
+        private static GameObject _functionControllerContainer;
+
+        // navigating these scenes in order triggers OnNewGame
+        private static readonly IList<string> _newGameDetectionScenes =
```

Comment:
These scene names may need to change for AI.  They have not caused issues yet.

Reply:
These names are almost definitely all different. They might not even be needed depending on how you hook things.